geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionVector.java

```diff
@@ -804,26 +811,34 @@ public long getVersionForMember(T mbr) {
    * @return true if this vector has seen the given version
    */
   public boolean contains(T id, long version) {
-    if (id.equals(this.myId)) {
-      if (isForSynchronization()) {
-        // a sync vector only has one holder & no valid version for the vector's owner
+    RegionVersionHolder<T> holder = this.memberToVersion.get(id);
+    // For region synchronization.
+    if (isForSynchronization()) {
+      if (holder == null) {
+        // we only care about missing changes from a particular member, and this
+        // vector is known to contain that member's version holder
         return true;
       }
+      if (id.equals(this.myId)) {
```

Comment:
this piece will not happen since the memberToVersion now contains myId.

Reply:
I think the region it won't happen is because a version source (as InternalDistributedMember) would not send region sync gii message for itself as a lost member. 

However for version as a diskstoreid, this still could happen, but it would also execute the holder.contains(version);

Will remove this block of code.

Reply:
I found this piece of code is still necessary, because your getCloneForTransmission(mbr) only handled diskstore case.

Reply:
I think it can be removed as I mentioned earlier -- an InternalDistributedMember(non disk id) can not/would not require region sync for itself, can it?

Reply:
A member can locally-destroy a region and then recreate it.  In that case other members will already have a version holder for it.  Does that require a sync?

Reply:
Only region sync invocation I saw from the code before this fix is triggered from memberDeparted event. So locally destroying a region should not trigger the region sync. However, I think not removing this piece of code would not cause any harm either.